### ISSUE TITLE
Incorrect overriding of private methods

### ISSUE THREAD

**Issue by romanmikhailov** [TiVo/intellij-haxe#311]
For example we have class
```haxe
class Foo
{
    private function forOverride():Void
    {    
    } 
}
```

When we create class `Bar` extending `Foo` and call override method menu (For windows Ctrl+O) and select method `forOverride` it paste with incorrect signature (`public override function private`).

```haxe
class Bar extends Foo
{
    public override private function forOverride():Void
    {
    }
}
```

Example:

![bug_with_private_override](https://cloud.githubusercontent.com/assets/1431183/8258194/c4b5e6fe-16bb-11e5-9be6-d2b87948736b.gif)

**Comment by EBatTiVo** [comment]
I see, we need to remove the 'private' keyword.

**Comment by romanmikhailov** [comment]
Why not leave visibility as-is?

**Comment by EBatTiVo** [comment]
I guess we could.  I just jumped to the conclusion that since a 'private' method was being overridden, making it public was what would be wanted.  I see your point.

**Comment by as3boyan** [comment]
Yeah, we do not encourage to use `private` keyword, but that's up to users to do it or not. So it's best to leave it, but overriding a private function seems a lil bit weird to me... But of course ability to do that and flexibility is great. Didn't thought about such use. Generate methods may have checks for `public` attribute, if it's not there - it will be added, I think... Anyway it shouldn't do it in this case.

**Comment by mayakwd** [comment]
Already fixed
